CondFormats/SiPixelTransient/interface/SiPixelTemplate.h

```diff
@@ -207,12 +207,12 @@ struct SiPixelTemplateStore {  //!< template storage structure
   SiPixelTemplateHeader head;
 #ifndef SI_PIXEL_TEMPLATE_USE_BOOST
   float cotbetaY[60];
-  float cotbetaX[5];
-  float cotalphaX[29];
+  float cotbetaX[60];
+  float cotalphaX[60];
   //!< 60 Barrel y templates spanning cluster lengths from 0px to +18px [28 entries for fpix]
   SiPixelTemplateEntry enty[60];
   //!< 29 Barrel x templates spanning cluster lengths from -6px (-1.125Rad) to +6px (+1.125Rad) in each of 5 slices [3x29 for fpix]
-  SiPixelTemplateEntry entx[5][29];
+  SiPixelTemplateEntry entx[60][60];
```

Comment:
What is the impact on memory? (my quick estimate would indicate that the `sizeof(SiPixelTemplateStore)` would be something like 7.4 MB, but I could have missed someting). Is an `SiPixelTemplateStore` object stored in stack anywhere? (just wondering if there is any risk to hit stack size issues like in #22458)

Reply:
@makortel do you have a specific memory check you would like to see? I am not sure if `SiPixelTemplateStore` is stored in stack. Maybe @mmusich is more familiar with this.

I ran the time and memory module comparison here:
**Before PR:**
`Summary for 10 events
Max VSIZ 6816.91 on evt 4 ; max RSS 3721.88 on evt 7
Time av 12.284 s/evt   max 26.0459 s on evt 4
CPU av 0 s/evt   max 0 s on evt -1
M1 Time av 11.3868 s/evt   max 26.0459 s on evt 4
M1 CPU av 0 s/evt   max 0 s on evt -1
M8 Time av 5.38321 s/evt   max 5.44807 s on evt 9`

**After PR:**
`Summary for 10 events
Max VSIZ 7344.61 on evt 6 ; max RSS 3759.03 on evt 7
Time av 16.4558 s/evt   max 33.244 s on evt 4
CPU av 0 s/evt   max 0 s on evt -1
M1 Time av 14.9319 s/evt   max 33.244 s on evt 4
M1 CPU av 0 s/evt   max 0 s on evt -1
M8 Time av 8.26217 s/evt   max 8.43981 s on evt 9`

I also ran the profile igprof, this shows the CPU use:
**After PR:**
https://cmantill.web.cern.ch/cmantill/cgi-bin//igprof-navigator/ig_phase2pr_Nov21_cpu/
or maybe more specifically:
https://cmantill.web.cern.ch/cmantill/cgi-bin//igprof-navigator/ig_phase2pr_Nov21_cpu/1283

And this shows the memory use (but I can't really tell anything from these logs so maybe you are more familiar navigating them):
**Before PR:**
https://cmantill.web.cern.ch/cmantill/cgi-bin//igprof-navigator/ig_phase2_nopr_Nov21
**After PR:**
https://cmantill.web.cern.ch/cmantill/cgi-bin//igprof-navigator/ig_phase2pr_Nov21

Reply:
> do you have a specific memory check you would like to see? 

Mainly if I estimated the object size correctly or not.

> I am not sure if `SiPixelTemplateStore` is stored in stack. Maybe @mmusich is more familiar with this.

According to dxr
https://cmssdt.cern.ch/dxr/CMSSW/search?q=%2Btype-ref%3ASiPixelTemplateStore
the only place where the object is in stack is in
https://github.com/cms-sw/cmssw/blob/9f9e834936b31e1c8fa77e358f8dd32de0f58420/CondFormats/SiPixelTransient/src/SiPixelTemplate.cc#L165
which I suppose is fine as that code path is not run within CMSSW. And I see the corresponding piece in the code path run within CMSSW
https://github.com/cms-sw/cmssw/blob/9f9e834936b31e1c8fa77e358f8dd32de0f58420/CondFormats/SiPixelTransient/src/SiPixelTemplate.cc#L734-L735
was already change to heap allocation in https://github.com/cms-sw/cmssw/pull/22458.

Good, sorry for the noise.